### PR TITLE
[Feature-Seller] 1. Update delete item 2. Update get all items

### PR DESCRIPTION
**Pull request by Alicewillbe**
1. delete item will not succeed if any appointment attached to it
2. get all items from db

### CODE REVIEW

File: src/main/java/com/spare4fun/core/dao/ItemDaoImpl.java [hunk 1]
```diff
@@ -65,6 +69,27 @@ public void deleteItemById(int itemId) {
...
+                    .where(builder.equal(root.get("item"), itemId));
+            appointments = session
+                    .createQuery(criteriaQuery)
+                    .getResultList();
+        } catch (NoResultException e) {
+            return false;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return true;
```
Comment (Alicewillbe) [inline src/main/java/com/spare4fun/core/dao/ItemDaoImpl.java:90]: Always return true s.t. we can never delete an item.
Here we should actually do ```return !appointements.isEmpty();```

File: src/main/java/com/spare4fun/core/dao/ItemDaoImpl.java [hunk 2]
```diff
@@ -65,6 +69,27 @@ public void deleteItemById(int itemId) {
...
+            CriteriaQuery<Appointment> criteriaQuery = builder.createQuery(Appointment.class);
+            Root<Appointment> root = criteriaQuery.from(Appointment.class);
+            criteriaQuery
+                    .select(root)
+                    .where(builder.equal(root.get("item"), itemId));
+            appointments = session
+                    .createQuery(criteriaQuery)
+                    .getResultList();
+        } catch (NoResultException e) {
+            return false;
```
Comment (Alicewillbe) [inline src/main/java/com/spare4fun/core/dao/ItemDaoImpl.java:86]: I don't think ```getResultList()``` will throw this exception.

File: src/main/java/com/spare4fun/core/dao/ItemDaoImpl.java [hunk 3]
```diff
@@ -65,6 +69,27 @@ public void deleteItemById(int itemId) {
...
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (Alicewillbe) [inline src/main/java/com/spare4fun/core/dao/ItemDaoImpl.java:88]: If an exception ever happens, we should always stop users to delete an item.
so we can ```throw e``` here again for ```ItemController``` to stop deleting.

**Review by Alicewillbe — COMMENTED** [review]
@Nina-gu please modify

**Review by Nina-gu — APPROVED** [review]

**Pull request merged**